text/0000-iri-v181-transaction-and-bundle/0000-iri-v181-transaction-and-bundle.md

```diff
@@ -0,0 +1,980 @@
+    /// between transactions by putting the hash of a transaction into its downstream neighbor.
+    ///
+    /// Proof of Work (PoW) allows your transactions to be accepted by the network. On the IOTA network, PoW is only
+    /// a rate control mechanism. After PoW, a bundle is ready to be sent to the network. Given a `trunk` and
+    /// a `branch` returned by [`getTransactionsToApprove`], the process iterates over each transaction of the bundle,
+    /// sets some attachment related fields and does individual PoW.
+    ///
+    /// [`getTransactionsToApprove`]: https://docs.iota.org/docs/node-software/0.1/iri/references/api-reference#gettransactionstoapprove
+    ///
+    /// TODO: This function relies on some `Sponge` and `ProofOfWork` traits, which are not yet defined. The code below lays out
+    ///       how proof of work is thought to be used.
+    pub fn chain_and_attach<S: Sponge, P: ProofOfWork>(
+        mut self,
+        trunk_tip: TransactionHash,
+        branch_tip: TransactionHash,
+        proof_of_work: P,
+        sponge: S,
+    ) -> Result<Self, OutgoingBundleError> {
+        unimplemented!()
+
+        // Iterates over all transactions starting from the head, setting trunks, branches, timestamps, tags, and nonce.
+        //
+        // NOTE: We do not provide an implementation here.
+        let builder_iterator = self.iter_from_head();
+
+        // Set the head transaction
+        let head_transaction = builder_iterator.next().ok_or("builder does not contain any transaction builders")?;
+        head_transaction.trunk(trunk_tip);
+        head_transaction.branch(branch_tip);
+
+        head_transaction.attachment_timestamp = timestamp();
+        head_transaction.attachment_timestamp_lower = 0;
+        head_transaction.attachment_timestamp_upper = 3812798742493;
+
+        if head_transaction.tag.is_empty() {
+            head_transaction.tag = transaction_draft.obsolete_tag;
+        }
+        head_transaction.set_nonce_from_proof_of_work(proof_of_work);
+
+        // `next_local_trunk` is the hash that will be used as value of the current transaction pointing to the previous
+        // one.
+        let mut next_local_trunk = head_transaction.calculate_hash(sponge);
+
+        // Set the rest of the transactions
+        for transaction_draft in builder_iterator {
+            transaction_draft.branch(trunk_tip);
+            transaction_draft.trunk(branch_tip);
+
+            transaction_draft.attachment_timestamp = timestamp();
+            transaction_draft.attachment_timestamp_lower = 0;
+            transaction_draft.attachment_timestamp_upper = 3812798742493;
+
+            if transaction_draft.tag.is_empty() {
+                transaction_draft.tag = transaction_draft.obsolete_tag;
+            }
+
+            transaction_draft.set_nonce_from_proof_of_work(proof_of_work);
+            next_local_trunk = transaction_draft.calculate_hash(sponge);
```

Comment:
You are setting `next_local_trunk` but not using it.

Reply:
Fixed.